### PR TITLE
CR-6453-k8s-resources-in-installed-git-source

### PR DESCRIPTION
**Pull request by elad-codefresh**

### CODE REVIEW

File: cmd/commands/git-source.go [hunk 1]
```diff
@@ -389,8 +551,11 @@ func RunGitSourceEdit(ctx context.Context, opts *GitSourceEditOptions) error {
 		return fmt.Errorf("failed to write the updated config.json of git-source: %s. Err: %w", opts.GsName, err)
 	}
 
-	log.G(ctx).Info("Pushing updated GitSource to the installation repo")
-	if err := apu.PushWithMessage(ctx, repo, fmt.Sprintf("Persisted an updated git-source '%s'", opts.GsName)); err != nil {
```
Comment (elad-codefresh) [inline cmd/commands/git-source.go]: I'm not sure why apu.PushWithMessage removal got included in this pr

File: cmd/commands/runtime.go [hunk 2]
```diff
@@ -441,9 +448,11 @@ func intervalCheckIsRuntimePersisted(ctx context.Context, runtimeName string, wg
 		}
 	}
 
-	stop()
+	if err != nil {
```
Comment (roi-codefresh) [inline cmd/commands/runtime.go]: error will never be nil here, look at the code.

File: cmd/commands/runtime.go [hunk 3]
```diff
@@ -441,9 +448,11 @@ func intervalCheckIsRuntimePersisted(ctx context.Context, runtimeName string, wg
...
+		return fmt.Errorf("failed to complete the runtime installation due to timeout. Error: %w", err)
+	}
 
-	return fmt.Errorf("timed out while waiting for runtime installation to complete")
+	return fmt.Errorf("failed to complete the runtime installation due to timeout")
```
Comment (roi-codefresh) [inline cmd/commands/runtime.go]: I think the other message is clearer

File: cmd/commands/runtime.go [hunk 4]
```diff
@@ -420,6 +426,7 @@ func intervalCheckIsRuntimePersisted(ctx context.Context, runtimeName string, wg
 	longetThanUsualMsg := waitMsg + " (this is taking longer than usual, you might need to check your cluster for errors)"
 	stop := util.WithSpinner(ctx, waitMsg)
 	ticker := time.NewTicker(time.Second * 10)
+	var err error
```
Comment (roi-codefresh) [inline cmd/commands/runtime.go]: what is the purpose for this?

File: cmd/commands/git-source.go [hunk 5]
```diff
@@ -163,32 +183,170 @@ func RunGitSourceCreate(ctx context.Context, opts *GitSourceCreateOptions) error
...
 
-		commitMsg := fmt.Sprintf("Created demo workflow template in %s Directory", opts.gsCloneOpts.Path())
+		pOpts := &git.PushOptions{
+			CommitMsg: fmt.Sprintf("Created demo workflow template in %s Directory", opts.GsCloneOpts.Path()),
+		}
 
-		log.G(ctx).Info("Pushing a demo workflow-template to the new git-source repo")
-		if err := apu.PushWithMessage(ctx, gsRepo, commitMsg); err != nil {
-			return fmt.Errorf("failed to push demo workflow-template to git-source repo: %w", err)
+		_, err = gsRepo.Persist(ctx, pOpts)
```
Comment (roi-codefresh) [inline cmd/commands/git-source.go]: we don't use this anymore, please switch to `apu.PushWithMessage` as it has built it retry mechanism

File: cmd/commands/git-source.go [hunk 6]
```diff
@@ -62,6 +72,14 @@ type (
 		InsCloneOpts *git.CloneOptions
 		GsCloneOpts  *git.CloneOptions
 	}
+
+	GitSourceCronExampleOptions struct {
```
Comment (roi-codefresh) [inline cmd/commands/git-source.go]: this needs to be private

File: cmd/commands/git-source.go [hunk 7]
```diff
@@ -163,32 +183,170 @@ func RunGitSourceCreate(ctx context.Context, opts *GitSourceCreateOptions) error
...
+			Template: &sensorsv1alpha1.Template{
+				ServiceAccountName: "argo-server",
+			},
+			Dependencies: dependencies,
+			Triggers:     triggers,
+		},
+	}, nil
+}
+
+func CreateCronExampleTrigger() (*sensorsv1alpha1.Trigger, error) {
```
Comment (roi-codefresh) [inline cmd/commands/git-source.go]: This needs to be private

File: cmd/commands/git-source.go [hunk 8]
```diff
@@ -291,7 +449,9 @@ func NewGitSourceDeleteCommand() *cobra.Command {
 		},
 	}
 
-	insCloneOpts = apu.AddCloneFlags(cmd, &apu.CloneFlagsOptions{})
+	insCloneOpts = git.AddFlags(cmd, &git.AddFlagsOptions{
```
Comment (roi-codefresh) [inline cmd/commands/git-source.go]: change it back to `apu.AddCloneFlags` please

File: cmd/commands/git-source.go [hunk 9]
```diff
@@ -355,14 +515,16 @@ func NewGitSourceEditCommand() *cobra.Command {
 		},
 	}
 
-	insCloneOpts = apu.AddCloneFlags(cmd, &apu.CloneFlagsOptions{
+	insCloneOpts = git.AddFlags(cmd, &git.AddFlagsOptions{
```
Comment (roi-codefresh) [inline cmd/commands/git-source.go]: change back

File: cmd/commands/git-source.go [hunk 10]
```diff
@@ -355,14 +515,16 @@ func NewGitSourceEditCommand() *cobra.Command {
...
 		CreateIfNotExist: true,
+		FS:               memfs.New(),
 	})
 
-	gsCloneOpts = apu.AddCloneFlags(cmd, &apu.CloneFlagsOptions{
+	gsCloneOpts = git.AddFlags(cmd, &git.AddFlagsOptions{
```
Comment (roi-codefresh) [inline cmd/commands/git-source.go]: change back

File: cmd/commands/git-source.go (hunk 1 above)
Comment (roi-codefresh) [reply cmd/commands/git-source.go]: Becase you didn't handle conflicts well, please fix.

File: cmd/commands/git-source.go [hunk 11]
```diff
@@ -62,6 +71,14 @@ type (
 		InsCloneOpts *git.CloneOptions
 		GsCloneOpts  *git.CloneOptions
 	}
+
+	gitSourceCronExampleOptions struct {
+		RuntimeName         string
```
Comment (roi-codefresh) [inline cmd/commands/git-source.go]: all the fields also need to be private

File: cmd/commands/git-source.go [hunk 12]
```diff
@@ -176,19 +201,138 @@ func RunGitSourceCreate(ctx context.Context, opts *GitSourceCreateOptions) error
...
 		return fmt.Errorf("failed to create git-source application. Err: %w", err)
 	}
 
-	log.G(ctx).Infof("Successfully created the git-source: '%s'", opts.gsName)
+	log.G(ctx).Infof("Successfully created the git-source: '%s'", opts.GsName)
+
+	return nil
+}
+
+func CreateCronExamplePipeline(opts *gitSourceCronExampleOptions) error {
```
Comment (roi-codefresh) [inline cmd/commands/git-source.go]: change to private

**Review by roi-codefresh — CHANGES_REQUESTED** [review]

File: cmd/commands/git-source.go [hunk 13]
```diff
@@ -176,19 +199,138 @@ func RunGitSourceCreate(ctx context.Context, opts *GitSourceCreateOptions) error
...
+	eventSourceFilePath := opts.gsFs.Join("resources", opts.eventSourceFileName)
+	sensorFilePath := opts.gsFs.Join("resources", opts.sensorFileName)
+
+	eventSource := &eventsourcev1alpha1.EventSource{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       eventsourcereg.Kind,
+			APIVersion: eventsourcereg.Group + "/v1alpha1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "calender",
```
Comment (roi-codefresh) [inline cmd/commands/git-source.go]: this is a typo, it's calendar I think

Comment (roi-codefresh) [reply cmd/commands/git-source.go]: search and replace this

File: pkg/util/events/util.go [hunk 14]
```diff
@@ -170,6 +177,12 @@ func CreateResourceEventSource(opts *CreateResourceEventSourceOptions) *eventsou
 	}
 }
 
+func CreateCalenderEventSource(opts *CreateCalenderEventSourceOptions) *eventsourcev1alpha1.CalendarEventSource {
```
Comment (roi-codefresh) [inline pkg/util/events/util.go]: Who is calling this?

File: cmd/commands/git-source.go [hunk 15]
```diff
@@ -176,19 +199,138 @@ func RunGitSourceCreate(ctx context.Context, opts *GitSourceCreateOptions) error
...
+					Resource: "workflows",
+				},
+				Operation: sensorsv1alpha1.Submit,
+				Source: &sensorsv1alpha1.ArtifactLocation{
+					Resource: &workflowResource,
+				},
+				Parameters: []sensorsv1alpha1.TriggerParameter{
+					{
+						Src: &sensorsv1alpha1.TriggerParameterSource{
+							DependencyName: "calender-dep",
```
Comment (roi-codefresh) [inline cmd/commands/git-source.go]: shouldn't these be from the `store`?

File: cmd/commands/git-source.go [hunk 16]
```diff
@@ -404,25 +546,35 @@ func createDemoWorkflowTemplate(gsFs fs.FS, runtimeName string) error {
 			APIVersion: wfv1alpha1.SchemeGroupVersion.String(),
 		},
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      "demo-workflow-template",
-			Namespace: runtimeName,
+			Name: "hello-world",
```
Comment (roi-codefresh) [inline cmd/commands/git-source.go]: This can be moved to `store`

File: pkg/util/events/util.go [hunk 17]
```diff
@@ -57,6 +57,10 @@ type (
 		TokenSecretName string
 	}
 
+	CreatecalendarEventSourceOptions struct {
```
Comment (roi-codefresh) [inline pkg/util/events/util.go]: delete this, this is unused

File: cmd/commands/runtime.go [hunk 18]
```diff
@@ -61,15 +61,17 @@ import (
...
+		RuntimeName         string
+		RuntimeToken        string
+		IngressHost         string
+		Insecure            bool
+		Version             *semver.Version
+		GsCloneOpts         *git.CloneOptions
+		InsCloneOpts        *git.CloneOptions
+		KubeFactory         kube.Factory
+		CommonConfig        *runtime.CommonConfig
+		SensorFileName      string
```
Comment (roi-codefresh) [inline cmd/commands/runtime.go]: no need for those two options, they are coming from the store

**Review by roi-codefresh — APPROVED** [review]

**Pull request merged**